app/uk/gov/hmrc/transitmovementsrouter/connectors/EISConnector.scala

```diff
@@ -108,11 +111,12 @@ class EISConnectorImpl(
           HeaderCarrier(requestChain = hc.requestChain, extraHeaders = hc.headers(Seq("X-Client-Id")))
         else HeaderCarrier(requestChain = hc.requestChain)
 
-      val requestId      = hc.requestId.map(_.value)
-      val correlationId  = UUID.randomUUID().toString
-      val accept         = MimeTypes.XML
-      val conversationId = ConversationId(movementId, MessageId("0000000000000000")) // ConversationId(movementId, messageId)
-      val date           = HTTP_DATE_FORMATTER.format(OffsetDateTime.now())
+      val requestId       = hc.requestId.map(_.value)
+      val correlationId   = UUID.randomUUID().toString
+      val accept          = MimeTypes.XML
+      val conversationId  = ConversationId(movementId, MessageId("0000000000000000")) // ConversationId(movementId, messageId)
+      val date            = HTTP_DATE_FORMATTER.format(OffsetDateTime.now())
+      val lrnRegexPattern = "The supplied LRN: ([a-zA-Z0-9]+) has already been used by submitter: ([a-zA-Z0-9]+)".r
```

Comment:
This variable should be a lazy val on the class level as this is a static regex that won't change. That way we calculate it once, rather than for each request.

Reply:
Yes agree, updated it.